cmd.go

```diff
@@ -137,59 +139,28 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
 	return r.prevPreRunE(cmd, args)
 }
 
-func (r *Runner) runE(runner interface{}, cfg interface{}) func(cmd *cobra.Command, args []string) error {
+func (r *Runner) runE(module fx.Option) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		// Unmarshal the configuration
-		if err := Unmarshal(r.prefix, cfg); err != nil {
-			return errors.Wrap(err, "unable to unmarshal configuration")
-		}
-
 		l := log.New()
 
 		loglevelFlag, err := cmd.Flags().GetString("loglevel")
 		if err != nil {
 			return err
 		}
 
-		if module, ok := runner.(fx.Option); ok {
-			if _, ok = cfg.(config.Config); ok {
-				runner = func(ctx context.Context, cfg config.Config) error {
-					app := fx.New(
-						logging.WithLogger(l),
-						FxContext(ctx),
-						FxOptions(cfg),
-						fx.Supply(cmd),
-						fx.Supply(service.Name(r.prefix)),
-						fx.Supply(logging.Level(loglevelFlag)),
-						Module,
-						module,
-					)
-
-					app.Run()
-
-					return app.Err()
-				}
-			} else {
-				panic("if providing an fx Module, then config must implement ProvidesFxOptions")
-			}
-		}
+		app := fx.New(
+			logging.WithLogger(l),
+			FxContext(cmd.Context()),
+			fx.Supply(cmd),
+			fx.Supply(service.Name(r.prefix)),
+			fx.Supply(logging.Level(loglevelFlag)),
+			module,
+			Module,
```

Comment:
Is it safe to re-order this? This to make sure that the `Load(..)` is invoked prior to all other invokes.

Reply:
No it is not safe because invoker hooks are registered in order.